Makefile.shared

```diff
@@ -311,3 +311,30 @@ partialclean::
 	rm -f driver/compdynlink.mli
 	rm -f driver/compdynlink.mlopt
 
+# The native toplevel
+
+compilerlibs/ocamlopttoplevel.cmxa: $(OPTTOPLEVEL:.cmo=.cmx)
+	$(CAMLOPT) -a -o $@ $(OPTTOPLEVEL:.cmo=.cmx)
+partialclean::
+	rm -f compilerlibs/ocamlopttoplevel.cmxa
+
+# When the native toplevel executable has an extension (e.g. ".exe"),
+# provide a phony 'ocamlnat' synonym
+
+ifneq ($(EXE),)
+.PHYNY: ocamlnat
```

Comment:
Did you try with the original rule targetting "ocamlnat" (even if it 
produces "ocamlnat.exe" in reality)?